### PR TITLE
Adding the checksums for verification

### PR DESCRIPTION
**Pull request by silverdaz**
The verification step now uses the checksum of the final files, and not the decryption step anymore.

On the side, we now also capture (in the logs) the error messages from GPG.

Note: Not sure that PR is correct, since I tried to split it in 2. Well...we'll see if the reviewers understand.

### CODE REVIEW

File: lega/crypto.py [hunk 1]
```diff
@@ -177,7 +192,7 @@ def ingest(enc_file,
              '--decrypt', enc_file
```
Comment (viklund) [inline lega/crypto.py]: I think that `code` should be called `cmd` instead, clearer that way that it's an external command to run.

File: lega/verify.py [hunk 2]
```diff
@@ -31,10 +31,12 @@ def work(data):
     '''Verifying that the file in the vault does decrypt properly'''
 
     file_id = data['file_id']
-    filename, org_hash, org_hash_algo, vault_filename = db.get_details(file_id)
+    filename, org_hash, org_hash_algo, vault_filename, vault_checksum = db.get_details(file_id)
 
-    crypto.decrypt_from_vault( vault_filename, org_hash, org_hash_algo )
+    #crypto.decrypt_from_vault( vault_filename, org_hash, org_hash_algo )
```
Comment (viklund) [inline lega/verify.py]: Remove code that's not needed anymore.

File: lega/worker.py [hunk 3]
```diff
@@ -77,11 +78,9 @@ def work(data):
     
     ################# Check integrity of encrypted file
     LOG.debug(f"Verifying the {hash_algo} checksum of encrypted file: {inbox_filepath}")
-    with open(inbox_filepath, 'rb') as inbox_file: # Open the file in binary mode. No encoding dance.
-        if not checksum(inbox_file, filehash, hashAlgo = hash_algo):
-            errmsg = f"Invalid {hash_algo} checksum for {inbox_filepath}"
-            LOG.warning(errmsg)
-            raise exceptions.Checksum(file_id, hash_algo, f'for {inbox_filepath}')
+    if not checksum(inbox_filepath, filehash, hashAlgo = hash_algo):
+        LOG.warning(f"Invalid {hash_algo} checksum for {inbox_filepath}")
```
Comment (viklund) [inline lega/worker.py]: Technically this should be `LOG.error`, right? In my mind warnings are things that go a little bit strange, but it's still possible to continue.

Might be more instances of this.

File: lega/crypto.py [hunk 4]
```diff
@@ -125,40 +127,53 @@ def __init__(self, hashAlgo, target_h, done):
...
         self.target_handler.write(b'\n')
+        self.target_digest.update(b'\n')
         LOG.debug('Writing key to file')
         self.target_handler.write(encryption_key)
+        self.target_digest.update(encryption_key)
         LOG.debug('Writing nonce to file')
         self.target_handler.write(nonce)
+        self.target_digest.update(nonce)
         self.engine = engine
+        # And now, daddy...
```
Comment (viklund) [inline lega/crypto.py]: I think you should have `__init__()` call two methods called `setup_digest(header, key, nonce)` and `setup_target(header, key, nonce)`.

File: lega/crypto.py (hunk 1 above)
Comment (silverdaz) [inline lega/crypto.py]: fixed

File: lega/verify.py (hunk 2 above)
Comment (silverdaz) [inline lega/verify.py]: ok då...

File: lega/worker.py (hunk 3 above)
Comment (silverdaz) [inline lega/worker.py]: fixed

**Comment by silverdaz** [comment]
I'll reshape the code to use generators or generator-like functions for the encryption/decryption in the next sprint

File: lega/crypto.py (hunk 4 above)
Comment (silverdaz) [inline lega/crypto.py]: Nja... I re-ordered the lines and added comments instead

File: lega/crypto.py [hunk 5]
```diff
@@ -112,53 +112,68 @@ class ReEncryptor(asyncio.SubprocessProtocol):
...
+        LOG.info(f'Starting the encrypting engine')
+        encryption_key, mode, nonce = next(self.engine)
 
-        encryption_key, mode, nonce = next(engine)
         self.header = make_header(CONF.getint('worker','active_key'), len(encryption_key), len(nonce), mode)
-        LOG.info(f'Writing header to file: {self.header}')
-        self.target_handler.write(self.header.encode('utf-8')) # includes \n
-        self.target_handler.write(b'\n')
-        LOG.debug('Writing key to file')
+        # includes \n
```
Comment (viklund) [inline lega/crypto.py]: ?? Just remove this.

**Pull request merged**